dialogue-clients/src/main/java/com/palantir/dialogue/clients/ReloadingClientFactory.java

```diff
@@ -223,12 +227,12 @@ public PerHostClientFactory perHost(String serviceName) {
         Preconditions.checkNotNull(serviceName, "serviceName");
         String channelName = ChannelNames.reloading(serviceName, params);
 
-        Refreshable<List<DialogueChannel>> perHostDialogueChannels = configurationForService(serviceName)
+        Refreshable<Map<PerHostTarget, Channel>> perHostChannels = configurationForService(serviceName)
```

Comment:
Slight preference to internally represent things like this using the immutables type for clarity
```suggestion
        Refreshable<ImmutableMap<PerHostTarget, Channel>> perHostChannels = configurationForService(serviceName)
```

This also has the minor upside of making code like 288 a bit simpler:
```diff
-ImmutableList.copyOf(channels.values())
+channels.values().asList()
```

Reply:
This runs into the problem mentioned below in https://github.com/palantir/dialogue/pull/2192#discussion_r1520152894.

You can't return a `Refreshable<ImmutableMap>` from a method that returns `Refreshable<Map>`.